include/deal.II/lac/lapack_templates.h

```diff
@@ -1326,6 +1348,145 @@ syevx (const char *, const char *, const char *, const types::blas_int *, float
 #endif
 
 
+// Template wrapper for LAPACK functions dsyevr and ssyevr
+template <typename number>
+inline void
+syevr(const char * /*jobz*/,
+      const char * /*range*/,
+      const char * /*uplo*/,
+      const types::blas_int * /*n*/,
+      number * /*A*/,
+      const types::blas_int * /*lda*/,
+      const number * /*vl*/,
+      const number * /*vu*/,
+      const types::blas_int * /*il*/,
+      const types::blas_int * /*iu*/,
+      const number * /*abstol*/,
+      types::blas_int * /*m*/,
+      number * /*w*/,
+      number * /*z*/,
+      const types::blas_int * /*ldz*/,
+      types::blas_int * /*isuppz*/,
+      number * /*work*/,
+      types::blas_int * /*lwork*/,
+      types::blas_int * /*iwork*/,
+      types::blas_int * /*liwork*/,
+      types::blas_int * /*info*/)
+{
+  Assert(false, ExcNotImplemented());
+}
+
+
+#ifdef DEAL_II_WITH_LAPACK
+inline void
+syevr(const char *jobz,
+      const char *range,
+      const char *uplo,
+      const types::blas_int *n,
+      double *A,
+      const types::blas_int *lda,
+      const double *vl,
+      const double *vu,
+      const types::blas_int *il,
+      const types::blas_int *iu,
+      const double *abstol,
+      types::blas_int *m,
+      double *w,
+      double *z,
+      const types::blas_int *ldz,
+      types::blas_int *isuppz,
+      double *work,
+      types::blas_int *lwork,
+      types::blas_int *iwork,
+      types::blas_int *liwork,
+      types::blas_int *info)
+{
+  dsyevr_(jobz,range,uplo,n,A,lda,vl,vu,il,iu,abstol,m,w,z,ldz,isuppz,work,lwork,iwork,liwork,info);
+}
+#else
```

Comment:
you don't need that else block here as it will call the general function above which already throws an error.